### DISCUSSION TITLE
Dynamic collections of components (of the same type) logically grouped/coupled

### DISCUSSION THREAD

**Discussion by AgustinRamiroDiaz** [bevyengine/bevy#10041] [Q&A] [answered]
I want to know what's the best way to handle dynamic collections of components that are from the same Rust type.

Example so we can talk in the same terms: I'm writing a multiplayer snake clone game. In this game, I currently use `SpriteBundle`  to make up the grid and each snake segment of body. The thing is that I want to handle the logic of each snake in a struct like
```
struct Snake {
    segments: VecDeque<SpriteBundle>,
}
```

But if I make `Snake` a component, then each `SpriteBundle` is not treated as a component and does not get rendered in the screen

I thought about putting each segment as a component, and mark it with some player tag and body number. Then I'd be able to query all the segments, group them, sort them and do the collection/queue logic. The thing is that this approach is not comfortable, it feels hacky and probably hard to maintain

I've also seen other people use `Resource`s, but if I want it to be a multiplayer game I won't be able to create multiple Snakes

What would you do?

**Comment by alice-i-cecile** [comment, marked as answer]
```rust
struct Snake {
    segments: VecDeque<Entity>,
}
```

You need each of the segments to be their own entity, so then they're rendered properly :) With this solution, you can record that and also have multiple snakes. Just be careful not to desynchronize this list and the spawning / despawning! Custom commands can help a lot there.

**Reply by AgustinRamiroDiaz** [reply]
Thanks a lot @alice-i-cecile ! I'm very grateful for your help :bow: 

Correct me if there's a better way: I had to `#[derive(Component)]` on `Snake`, and spawn the `Snake` into the world so then my systems could query for them 

To close the loop, here's the working code I have at the moment for anyone encountering this question in the future
https://github.com/AgustinRamiroDiaz/snake_bevy/blob/5c1e28304608fe3494fbb3ffe1961233322f0b97/src/main.rs

**Reply by alice-i-cecile** [reply]
Yep that's exactly the idea :)
